src/ahbicht/validation/validation_results.py

```diff
@@ -168,3 +168,70 @@ def deserialize(self, data, **kwargs) -> ValidationResultInContext:
         so that the discriminator is the key and the validation_result the value.
         """
         return ValidationResultInContext(**data)
+
+
+@attrs.define(auto_attribs=True, kw_only=True)
+class ListOfValidationResultInContext:
+    """
+    Class to set validation result in context, for example with its discriminator.
+    """
+
+    validation_results: List[ValidationResultInContext] = attrs.field(
+        validator=attrs.validators.deep_iterable(
+            member_validator=attrs.validators.instance_of(ValidationResultInContext),
+            iterable_validator=attrs.validators.instance_of(list),
+        )
+    )
+
+    def replace_edi_seed_paths_with_bo4e_paths(self, edi_seed_to_bo4e_mappings: Dict[str, str]) -> None:
+        """
+        replaces the discriminators in the list of validation results with the respective values from the edi_seed,
+        if present
+        """
+        for validation_result in self.validation_results:
+            if validation_result.discriminator.startswith("$"):
+                edi_seed_path = validation_result.discriminator
+                if edi_seed_path in edi_seed_to_bo4e_mappings:
+                    bo4e_path = edi_seed_to_bo4e_mappings[edi_seed_path]
+                    validation_result.discriminator = bo4e_path
+
+    def filter_for_data_element_validation_results(self) -> None:
+        """
+        Returns only the DataElementValidationResults of a list of ValidationResultInContext
+        """
+        only_data_element_validation_results = []
+        for validation_result_in_context in self.validation_results:
+            if isinstance(validation_result_in_context.validation_result, DataElementValidationResult):
+                only_data_element_validation_results.append(validation_result_in_context)
+
+        self.validation_results = only_data_element_validation_results
```

Comment:
warum nicht direkt mit filter?
```suggestion
        self.validation_results = list(filter(lambda vr: isinstance(vr.validation_result, DataElementValidationResult) , self.validation_results))
```

https://docs.python.org/3/library/functions.html?highlight=filter#filter

Reply:
411af6c3ba133d6afa2f04f6bef1c32fa775f74f